src/doc.cpp

```diff
@@ -1571,12 +1571,72 @@ void Doc::SetChildZones() {
         if (fi->HasFacs()) {
             fi->SetZone(m_facsimile->FindZoneByUuid(fi->GetFacs()));
         }
-        /*
-        else {
-            LogError("If facsimilies are present then element %s should have it.", (*iter)->GetClassName().c_str());
-            assert(false);
+
+        // add default boundingboxes to the syls if the option is true
+        else if (!fi->HasFacs() && (*iter)->Is(SYL) && this->GetOptions()->m_createDefaultSylBBox.GetValue()) {
+            Zone *zone = new Zone();
+
+            // if it's syllable parent has position values just use those
+            FacsimileInterface *syllableFi = nullptr;
+            if ((*iter)->GetFirstParent(SYLLABLE)->GetFacsimileInterface()->HasFacs()) {
+                syllableFi = (*iter)->GetFirstParent(SYLLABLE)->GetFacsimileInterface();
+                Zone *tempZone = dynamic_cast<Zone *>(syllableFi->GetZone());
+                zone->SetUlx(tempZone->GetUlx());
+                zone->SetUly(tempZone->GetUly());
+                zone->SetLrx(tempZone->GetLrx());
+                zone->SetLry(tempZone->GetLry());
+            }
+
+            // otherwise get a boundingbox that comprises all the neumes in the syllable
+            else {
+                ArrayOfObjects children;
+                InterfaceComparison comp(INTERFACE_FACSIMILE);
+                (*iter)->GetFirstParent(SYLLABLE)->FindAllChildByComparison(&children, &comp);
```

Comment:
This should be broken into two parts with an assert statement added. i.e.
```c++
Syllable *parentSyllable = dynamic_cast<Syllable *>((*iter)->GetFirstParent(SYLLABLE));
assert(parentSyllable);
parentSyllable->FindAllChildByComparison(&children, &comp);
```
This will help catch any bugs that may happen in the future.